### PR TITLE
asserts: authority-id and brand-id of serial must match

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: asserts/device_asserts.go [hunk 1]
```diff
@@ -105,11 +105,22 @@ func (mod *Model) checkConsistency(db RODatabase, acck *AccountKey) error {
 // sanity
 var _ consistencyChecker = (*Model)(nil)
 
+func checkBrandAndAuthority(a Assertion) error {
```
Comment (emgee) [inline asserts/device_asserts.go]: maybe checkBranchMatchesAuthority?

**Comment by emgee** [comment]
+1, with small suggestion for name of new checker.

File: asserts/device_asserts.go [hunk 2]
```diff
@@ -105,11 +105,22 @@ func (mod *Model) checkConsistency(db RODatabase, acck *AccountKey) error {
...
+	return nil
+}
+
 var modelMandatory = []string{"core", "architecture", "gadget", "kernel", "store", "class"}
 
 func assembleModel(assert assertionBase) (Assertion, error) {
-	if assert.headers["brand-id"] != assert.headers["authority-id"] {
-		return nil, fmt.Errorf("authority-id and brand-id must match, model assertions are expected to be signed by the brand: %q != %q", assert.headers["authority-id"], assert.headers["brand-id"])
+	err := checkBrandAndAuthority(&assert)
+	if err != nil {
```
Comment (mvo5) [inline asserts/device_asserts.go]: (nitpick) this could be in a single line `if err := check...; err != nil {`

File: asserts/device_asserts.go [hunk 3]
```diff
@@ -171,7 +182,10 @@ func (ser *Serial) Timestamp() time.Time {
 // TODO: implement further consistency checks for Serial but first review approach
 
 func assembleSerial(assert assertionBase) (Assertion, error) {
-	// TODO: authority-id can only == canonical or brand-id
+	err := checkBrandAndAuthority(&assert)
```
Comment (mvo5) [inline asserts/device_asserts.go]: (nitpick) Same here, could also be one less line.

**Comment by mvo5** [comment]
:+1: (some minor nitpicking inside, but not really relevant)

File: asserts/device_asserts.go (hunk 2 above)
Comment (pedronis) [inline asserts/device_asserts.go]: @mvo5 I think you had this nitpick already a couple of times, should I really switch to that style,  it seems not to be really in my fingers

**Pull request merged**